app/models/shelter.rb

```diff
@@ -31,4 +31,12 @@ def alphabetical_pets
   def shelter_pets_filtered_by_age(age_filter)
     adoptable_pets.where("age >= ?", age_filter)
   end
+
+  def self.order_by_reverse_alpha
+    find_by_sql("SELECT * FROM shelters ORDER BY name DESC")
+  end
+
+  def self.find_pending_applications
+    joins(pets: { application_pets: :application }).where(applications: { status: 'Pending' }).distinct
+  end
```

Comment:
This is a great query, I see that you needed to get to the applications table through application_pets and you made use of your pets association to do that. An easier way to do this would be to add another association to the model:

```erb
has_many :application_pets, through: :pets
has_many :applications, through: :application_pets
```

And with this, you can simplify your query:

```ruby
def self.find_pending_applications
  applications.where(applications: { status: 'Pending' }).distinct
end
```